feat: Shows link location and respective icon in /bookings

kart1ka opened the pull request:
## What does this PR do?

Shows link location and respective icon in /bookings.
Fixes #11730 
/claim #11730 

![Screenshot from 2023-12-13 07-25-10](https://github.com/calcom/cal.com/assets/41051387/d40acff7-4470-4e59-bc70-1da8039725c5)


## Type of change

<!-- Please delete bullets that are not relevant. -->

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

Once a booking is created, go to the /bookings page and observe the link shown in booking card.

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

kart1ka committed:
Add metadata to bookingMinimalSelect

kart1ka committed:
add: Show link location in /bookings

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/calcom/cal.com?pullRequest=12760) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/calcom/cal.com?pullRequest=12760) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/calcom/cal.com?pullRequest=12760) it.</sub>

kart1ka committed:
Merge branch 'main' into meeting-link

kart1ka committed:
Merge branch 'calcom:main' into meeting-link

Udit-takkar commented:
@kart1ka the bounty tag was removed from the issue. not sure why i see algora bot comment here

kart1ka committed:
Merge branch 'main' into meeting-link

Udit-takkar commented on apps/web/components/booking/BookingListItem.tsx:
```diff
@@ -359,6 +376,33 @@ function BookingListItem(booking: BookingItemProps) {
+                      rel="noreferrer"
+                      className="text-sm leading-6 text-blue-600 hover:underline">
+                      <div className="flex items-center gap-2">
+                        <img
+                          src={provider?.iconUrl || ""}
+                          className="h-4 w-4 rounded-sm"
+                          alt={`${provider?.label} logo`}
+                        />
```
```suggestion
                     {provider?.iconUrl && (
                          <img
                            src={provider.iconUrl}
                            className="h-4 w-4 rounded-sm"
                            alt={`${provider?.label} logo`}
                          />
                        )}
```

Udit-takkar commented on apps/web/components/booking/BookingListItem.tsx:
```diff
@@ -93,6 +98,18 @@ function BookingListItem(booking: BookingItemProps) {
   const paymentAppData = getPaymentAppData(booking.eventType);
 
+  const location = booking.location as ReturnType<typeof getEventLocationValue>;
+  const locationVideoCallUrl: string | undefined = bookingMetadataSchema.parse(
+    booking?.metadata || {}
+  )?.videoCallUrl;
```
NIT
```suggestion
  const locationVideoCallUrl = bookingMetadataSchema.parse(
    booking?.metadata || {}
  )?.videoCallUrl;
```

Udit-takkar reviewed (dismissed):
We only want to display the url location and not address (organizer address or attendee address)

<img width="1259" alt="Screenshot 2023-12-14 at 12 57 40 AM" src="https://github.com/calcom/cal.com/assets/53316345/d85c5d98-208e-46af-bc7d-711dfe76d50d">

kart1ka committed:
Merge branch 'main' into meeting-link

kart1ka committed:
Refactor: Update variable declaration and conditional rendering in booking metadata

- Remove explicit type declaration in locationVideoCallUrl assignment
- Use conditional rendering for provider icon based on iconUrl existence

Co-authored-by: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>

kart1ka committed:
Display URL location exclusively, omitting addresses

kart1ka commented:
@Udit-takkar I have made the changes as per the request. Do let me know if any more changes need to be made.

PeerRich committed:
Merge branch 'main' into meeting-link

kart1ka commented:
Hi @Udit-takkar .  Is there any more changes that I need to make to get this PR merged?

Udit-takkar committed:
Merge branch 'main' into meeting-link
